forc-plugins/forc-doc/src/doc/descriptor.rs

```diff
@@ -310,4 +310,62 @@ impl Descriptor {
             _ => Ok(Descriptor::NonDocumentable),
         }
     }
+
+    /// Decides whether a [TypeInfo] is [Descriptor::Documentable] and returns a [Document] if so.
+    pub(crate) fn from_type_info(
+        type_info: &TypeInfo,
+        engines: &Engines,
+        module_info: ModuleInfo,
+    ) -> Result<Self> {
+        // Only primitive types will result in a documentable item. All other type documentation should come
+        // from the a declaration. Since primitive types do not have sway declarations, we can only generate
+        // documentation from their implementations.
+        let item_name = Ident::new_no_span(format!("{}", engines.help_out(type_info)));
+        // Build a fake module info for the primitive type.
+        let module_info = ModuleInfo {
+            module_prefixes: vec![module_info.project_name().into()],
+            attributes: None,
+        };
+        // TODO: Find a way to add descriptions without hardcoding them.
```

Comment:
Probably worth opening an issue for this so we don't forget.

Reply:
https://github.com/FuelLabs/sway/issues/6066